src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -416,11 +414,12 @@ internal EmbeddedText TryReadEmbeddedFileContent(string filePath, DiagnosticBag
             }
         }
 
-        private ImmutableArray<EmbeddedText> AcquireEmbeddedTexts(Compilation compilation, DiagnosticBag diagnostics)
+        private ImmutableArray<EmbeddedText?> AcquireEmbeddedTexts(Compilation compilation, DiagnosticBag diagnostics)
```

Comment:
It seems really odd that we explicitly allow for `null` values in this collection but we clearly do (see comment at the bottom of this method). Had thoughts about undoing that but felt it was too much for this PR.

Reply:
> It seems really odd that we explicitly allow for null values in this collection ...

Agreed, this seems odd. Especially since the collection is passed to other types (such as to `Compilation.CheckOptionsAndCreateModuleBuilder()`).

Could we just replace the `ImmutableArray.CreateBuilder<EmbeddedType?>(...)` with `ArrayBuilder.CreateInstance<EmbeddedText>()` and avoid adding to that builder if `text is null`?

---
In reply to: [502079305](https://github.com/dotnet/roslyn/pull/48460#discussion_r502079305) [](ancestors = 502079305)